packages/drivers/odsp-driver/src/snapshotFormatConverter.ts

```diff
@@ -0,0 +1,279 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert, stringToBuffer } from "@fluidframework/common-utils";
+import { IBlob, ISnapshotTree } from "@fluidframework/protocol-definitions";
+import { ISequencedDeltaOpMessage } from "./contracts";
+import { BlobCore, NodeCore, ReadBuffer, TreeBuilder } from "./snapshotRepresentation";
+
+const minReadVersion = "1.0";
+
+/**
+ *  Header of the downloaded snapshot.
+ */
+export interface ISnapshotHeader {
+    // This is the minimum version of the reader required to read the wire format of a given snapshot.
+    minReadVersion: string,
+    // Represents the version with which snapshot is created.
+    createVersion: string,
+    // Seq number at which the snapshot is created.
+    snapshotSeqNumber: number,
+}
+
+/**
+ * Class used to convert snapshot formats. It can convert binary compact representation to
+ * odsp snapshot and vice-versa.
+ */
+export class SnapshotFormatConverter {
+    constructor() {}
+
+    /**
+     * Recreates header section and then validate that.
+     * @param node - tree node to read header section from
+     */
+    private readAndValidateHeaderSection(node: NodeCore) {
+        assert(node.getString(0) === "MinReadVersion", "MinReadVersion header should be present");
```

Comment:
We should probably allow properties to be written in any order. I.e. have generic code that can read a property bag, and then this code that checks that certain properties are available.

Reply:
Replaced it with switch.